docs: Add nextflow auth reference docs (#3)
ewels/nextflow: docs/reference/cli.md

## Commands

(cli-auth)=

### `auth`

:::{versionadded} 25.09.0-edge
:::

Manage Seqera Platform authentication.

**Usage**

```console
$ nextflow auth <subcommand> [options]
```

**Description**

> The `auth` command provides authentication and configuration management for Seqera. It supports OAuth2 authentication for Seqera Cloud and Personal Access Token (PAT) authentication for Seqera Enterprise installations. Credentials are saved to `~/.nextflow/seqera_auth.config`.

Reviewer: Whilst technically true, Seqera Cloud auth's with OAuth2 and then creates a PAT and saves that locally. So it's still using PATs for Cloud (for now).

Author: New text is:

> The `auth` command provides authentication and configuration management for Seqera. For Seqera Cloud, it uses an OAuth2 authentication flow generate and save a Personal Access Token (PAT) locally. For Seqera Enterprise installations, it uses direct PAT authentication. Credentials are saved to `~/.nextflow/seqera_auth.config`.

> The `auth` command provides authentication and configuration management for Seqera. It supports OAuth2 authentication for Seqera Cloud and Personal Access Token (PAT) authentication for Seqera Enterprise installations. Credentials are saved to `~/.nextflow/seqera_auth.config`.

Author: ```suggestion
The `auth` command provides authentication and configuration management for Seqera. For Seqera Cloud, an OAuth2 authentication flow is used to generate and save a Personal Access Token (PAT) locally. For Seqera Enterprise installations, it uses direct PAT authentication. Credentials are saved to `~/.nextflow/seqera_auth.config`.
```

**Options**

`-h, -help`
: Prints the command usage information.

`-u, -url`

> : Specifies your Seqera API endpoint (default: `https://api.cloud.seqera.io`)

Reviewer: I don't think that this is top level, it's only for `auth login`

Author: Removing from here and listing under `login`

> 
> `-u, -url`
> : Specifies your Seqera API endpoint (default: `https://api.cloud.seqera.io`)

Author: ```suggestion
```

**Subcommands**

`login`

> : Authenticates with Seqera, saves credentials, and configures monitoring and workspaces.

Reviewer: Also the primary compute environment for the workspace, as of last night. But maybe too much detail.

Author: New text is:

> `login`
: Authenticates with Seqera and saves credentials. Sets Seqera primary compute environment, monitoring, and workspace.
: The following options are available:
  `-u, -url`
  : Specifies your Seqera API endpoint (default: `https://api.cloud.seqera.io`)

`logout`

> : Removes Seqera authentication and revokes access token.

Reviewer: > revokes access token

For Cloud only, not enterprise

Author: New text is:

> `logout`
: Removes Seqera authentication and revokes the Seqera Cloud access token (if applicable).

`config`

> : Sets Seqera monitoring, workspace, and workspace.

Reviewer: As above, + primary CE now

Author: New text is:

> `config`
: Sets Seqera primary compute environment, monitoring, and workspace.

`status`
: Shows Seqera authentication status and configuration.

**Examples**

Authenticate with Seqera Cloud:

```console
$ nextflow auth login 
```

Authenticate with an Enterprise installation:

```console
$ nextflow auth login -u https://tower.example.com/api
```

View current authentication status:

```console

> $ nextflow auth status

Reviewer: I assume we can't have nice screenshots or example outputs?

Author: Not for these reference docs 😅 

A guide would be okay. However, we could add the example outputs as code. Let me know if you want to go that direction.

## Commands

(cli-auth)=

### `auth`

:::{versionadded} 25.09.0-edge
:::

Manage Seqera Platform authentication.

**Usage**

```console
$ nextflow auth <subcommand> [options]
```

**Description**

The `auth` command provides authentication and configuration management for Seqera. For Seqera Cloud, it uses an OAuth2 authentication flow generate and save a Personal Access Token (PAT) locally. For Seqera Enterprise installations, it uses direct PAT authentication. Credentials are saved to `~/.nextflow/seqera_auth.config`.

**Options**

`-h, -help`
: Prints the command usage information.

**Subcommands**

`login`
: Authenticates with Seqera and saves credentials. Sets Seqera primary compute environment, monitoring, and workspace.
: The following options are available:

>   `-u, -url`
>   : Specifies your Seqera API endpoint (default: `https://api.cloud.seqera.io`)

Reviewer: Incorrect indentation?

```suggestion
`-u, -url`
: Specifies your Seqera API endpoint (default: `https://api.cloud.seqera.io`)
```
